omod/src/main/java/org/openmrs/module/formentryapp/page/controller/forms/ExtensionPageController.java

```diff
@@ -54,6 +55,9 @@ public String post(@BindParams("extensionForm") ExtensionForm extensionForm,
 		}
 		
 		extensionForm.copyTo(extension);
+		if (StringUtils.isNotBlank(extension.getRequire())) {
+			extension.setRequire(StringEscapeUtils.unescapeHtml4(extension.getRequire()));
```

Comment:
@dkayiwa let me know if you have any concerns about this or better suggestions.  This has primarily manifested with & characters, if someone has a require expression like `expression1 && expression2`, then this ends up in the DB as `expression1 &amp;&amp; expression2`, which appframework fails on.

Reply:
What do you think about unescaping at the point when it is going to be used by the processing engine, instead of when it is just going to be displayed in the user interface?

Reply:
To be clear - this isn't "when it is just going to be displayed".  This is unescaping prior to saving in the DB.